src/GLib/signals.jl

```diff
@@ -370,16 +370,34 @@ function g_timeout_add(interval::Integer, cb::Function, user_data::CT) where CT
         0, UInt32(interval), callback, ref, deref)
 end
 
+_g_timeout_add_callback(cb::Function) = Cint(cb())
+function g_timeout_add(interval::Integer, cb::Function)
+    callback = @cfunction(_g_timeout_add_callback, Cint, (Ref{Function},))
+    ref, deref = gc_ref_closure(cb)
+    return ccall((:g_timeout_add_full, libglib),Cint,
+        (Cint, UInt32, Ptr{Cvoid}, Ptr{Cvoid}, Ptr{Cvoid}),
+        0, UInt32(interval), callback, ref, deref)
+end
+
 function g_idle_add(cb::Function, user_data::CT) where CT
 
     callback = @cfunction($cb, Cint, (Ref{CT},) )
     ref, deref = gc_ref_closure(user_data)
 
-    return ccall((:g_idle_add_full , libglib),Cint,
+    return ccall((:g_idle_add_full, libglib),Cint,
         (Cint, Ptr{Nothing}, Ptr{Nothing}, Ptr{Nothing}),
         0, callback, ref, deref)
 end
 
+_g_idle_add_callback(cb::Function) = Cint(cb())
```

Comment:
You might as well just define a single `_g_callback(cb::Function)` method and use it for both `idle_add` and `timeout_add`, since they are identical.